OpenRA.Mods.Common/Projectiles/LaserZap.cs

```diff
@@ -74,6 +81,21 @@ public class LaserZapInfo : IProjectileInfo
 
 		[PaletteReference] public readonly string HitAnimPalette = "effect";
 
+		[Desc("Image containing launch effect sequence.")]
+		public readonly string LaunchEffectImage = null;
+
+		[Desc("Launch effect sequence to play.")]
+		[SequenceReference("LaunchEffectImage")] public readonly string LaunchEffectSequence = null;
+
+		[Desc("Palette to use for launch effect.")]
+		[PaletteReference] public readonly string LaunchEffectPalette = "effect";
+
+		[Desc("Total time-frame in ticks over which the launch effect is spawned every LaunchEffectInterval.")]
+		public readonly int LaunchEffectDuration = 1;
+
+		[Desc("Launch effect is spawned every this many ticks during LaunchEffectDuration.")]
+		public readonly int LaunchEffectInterval = 1;
```

Comment:
I still don't understand the purpose of these two fields - it doesn't make sense to me why you would want to repeatedly spawn the same effect for only part of a weapon's firing duration.  If this is to implement a warmup-loop-cooldown cycle (like `fire-start` and `fire-loop` on the oil pump destruction) then that would be cleaner and certainly better for performance to define these explicitly in the `LaunchEffect`.

A demonstration/testcase for this would help a lot.

Reply:
I could do that, but - apart from not being easy to set up (don't have any suitable example in shp(td) or shp(ts) format) - by now I suspect it would get rejected anyway, for being aimed at a too specific scenario.

The use-case in my mod is that the laser muzzle effect is too "weak" with higher translucency, and still so-so (and too pixelated/aliased!) with very low or disabled translucency, so just prolonging the sequence doesn't cut it.
I tried all kinds of blend-modes, but using higher translucency + spawning another muzzle every tick for the `LaunchEffectDuration` is still by far producing the best-looking result (I even use a `LaunchEffectIntensity` that allows to spawn more than one effect per tick, which I didn't bother submitting as it's even more limited-scope).

I admit I can't think of any real uses outside that very specific use-case, it was more a case of "_might as well submit them upstream along with the generally useful stuff, to reduce my local diff and in case anyone else has a use for this_".
It's probably better to just drop them from this PR (I have to use a custom engine anyway, due to other changes).

Reply:
Ah, I see.  So you're relying on stacking many overlapping sprites in order to build up the strength of the effect?
We had the same problem in D2K, which I worked around by adding the `PaletteFromScaledPalette` trait to increase the contrast in the palette itself - mimicking  the effect of stacking multiple sprites but without the stacking.

Reply:
Awesome, I'm glad we could resolve this in a way that improves things for everyone.